src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java

```diff
@@ -120,14 +121,17 @@ private void applyDecision(CaseDetails caseDetails, JsonNode changeOrganisationR
         JsonNode organisationToRemoveNode = changeOrganisationRequestField.get(ORGANISATION_TO_REMOVE);
         Organisation organisationToAdd = objectMapper.convertValue(organisationToAddNode, Organisation.class);
         Organisation organisationToRemove = objectMapper.convertValue(organisationToRemoveNode, Organisation.class);
+        JsonNode createdBy = changeOrganisationRequestField.get(CREATED_BY);
 
         List<CaseUserRole> existingCaseAssignments = dataStoreRepository
             .getCaseAssignments(singletonList(caseDetails.getId()), null);
 
         if (organisationToAdd == null || isNullOrEmpty(organisationToAdd.getOrganisationID())) {
+            ((ObjectNode) orgPolicyNode).putIfAbsent("LastNoCRequestedBy", null);
             applyRemoveRepresentationDecision(existingCaseAssignments, caseRoleId, orgPolicyNode, organisationToRemove,
                 caseDetails.getId());
         } else {
+            ((ObjectNode) orgPolicyNode).putIfAbsent("LastNoCRequestedBy", createdBy);
```

Comment:
Similar comments to above - should probably sit in the `applyAddOrReplaceRepresentationDecision()` method, and `putIfAbsent()` presumably won't actually update it if there's an existing one (but we do want it to)?